File: src/java/com/portalpostal/dao/handler/LancamentoConciliadoHandler.java
package com.portalpostal.dao.handler;

import com.portalpostal.model.CentroCusto;
import com.portalpostal.model.Conta;
import com.portalpostal.model.Lancamento;
import com.portalpostal.model.LancamentoConciliado;
import com.portalpostal.model.PlanoConta;
import com.portalpostal.model.dd.TipoLancamento;
import java.sql.ResultSet;
import java.sql.SQLException;
import org.sql2o.ResultSetHandler;

public class LancamentoConciliadoHandler extends GenericHandler implements ResultSetHandler<LancamentoConciliado> {
        
    public LancamentoConciliadoHandler() {
        super("lancamento_conciliado");
    }
    
    public LancamentoConciliadoHandler(String table) {
        super(table);
    }

    public LancamentoConciliado handle(ResultSet result) throws SQLException {
        LancamentoConciliado lancamentoConciliado = new LancamentoConciliado();
        lancamentoConciliado.setIdLancamentoConciliado(getInt(result, "idLancamentoConciliado"));
        lancamentoConciliado.setConta(getConta(result));
        lancamentoConciliado.setPlanoConta(getPlanoConta(result));
        lancamentoConciliado.setCentroCusto(getCentroCusto(result));
        lancamentoConciliado.setLancamento(getLancamento(result));
        lancamentoConciliado.setTipo(TipoLancamento.values()[getInt(result, "tipo")]);
        lancamentoConciliado.setNumeroLote(getInt(result, "numeroLote"));
        lancamentoConciliado.setDataCompetencia(getDate(result, "dataCompetencia"));
        lancamentoConciliado.setDataEmissao(getDate(result, "dataEmissao"));
        lancamentoConciliado.setDataLancamento(getDate(result, "dataLancamento"));
        lancamentoConciliado.setValor(getDouble(result, "valor"));
        lancamentoConciliado.setHistorico(getString(result, "historico"));
        lancamentoConciliado.setUsuario(getString(result, "usuario"));
        return lancamentoConciliado;
    }
    
    private Conta getConta(ResultSet result) throws SQLException {
        if(!existColumn(result, "conta.idConta")) return null;
        return new ContaHandler().handle(result); 
    }
    
    private PlanoConta getPlanoConta(ResultSet result) throws SQLException {
        if(!existColumn(result, "plano_conta.idPlanoConta")) return null;
        return new PlanoContaHandler().handle(result); 
    }
    
    private CentroCusto getCentroCusto(ResultSet result) throws SQLException {
        if(!existColumn(result, "centro_custo.idCentroCusto")) return null;
        return new CentroCustoHandler().handle(result); 
    }
    
    private Lancamento getLancamento(ResultSet result) throws SQLException {
        if(!existColumn(result, "lancamento.idLancamento")) return null;
        if(!existFKValue(result, "lancamento.idLancamento")) return null;
        return new LancamentoHandler().handle(result); 
    }
    
}

File: web/app/js/handler/lancamentoConciliadoHandler.js
'use strict';

app.factory('LancamentoConciliadoHandler', function() {
    
    var handleList = function(lancamentoConciliadoList) { 
        return _.map(lancamentoConciliadoList, function(lancamentoConciliado) {
            return handle(lancamentoConciliado);
        });
    };
    
    var handle = function(lancamentoConciliado) {  
        var lancamentoConciliadoHandle = {};
        lancamentoConciliadoHandle.idLancamentoConciliado = getId(lancamentoConciliado);            
        lancamentoConciliadoHandle.conta = getConta(lancamentoConciliado);  
        lancamentoConciliadoHandle.planoConta = getPlanoConta(lancamentoConciliado);  
        lancamentoConciliadoHandle.centroCusto = getCentroCusto(lancamentoConciliado);
        lancamentoConciliadoHandle.lancamento = getLancamento(lancamentoConciliado);
        lancamentoConciliadoHandle.tipo = getTipo(lancamentoConciliado); 
        lancamentoConciliadoHandle.numeroLote = getNumeroLote(lancamentoConciliado);
        lancamentoConciliadoHandle.dataCompetencia = getDataCompetencia(lancamentoConciliado);
        lancamentoConciliadoHandle.dataEmissao = getDataEmissao(lancamentoConciliado);
        lancamentoConciliadoHandle.dataLancamento = getDataLancamento(lancamentoConciliado);
        lancamentoConciliadoHandle.valor = getValor(lancamentoConciliado);
        lancamentoConciliadoHandle.historico = getHistorico(lancamentoConciliado);
        lancamentoConciliadoHandle.usuario = getUsuario(lancamentoConciliado);
        return lancamentoConciliadoHandle;
    };
    
    var getConta = function(lancamentoConciliado) {
        if(!lancamentoConciliado.conta) return lancamentoConciliado.conta;
        return { idConta: lancamentoConciliado.conta.idConta };
    };
    
    var getPlanoConta = function(lancamentoConciliado) {
        if(!lancamentoConciliado.planoConta) return lancamentoConciliado.planoConta;    
        return { idPlanoConta: lancamentoConciliado.planoConta.idPlanoConta }; 
    };
    
    var getCentroCusto = function(lancamentoConciliado) {
        if(!lancamentoConciliado.centroCusto) return lancamentoConciliado.centroCusto;    
        return { idCentroCusto: lancamentoConciliado.centroCusto.idCentroCusto }; 
    };
    
    var getLancamento = function(lancamentoConciliado) {
        if(!lancamentoConciliado.lancamento) return lancamentoConciliado.lancamento;
        return { idLancamentoProgramado: lancamentoConciliado.lancamento.idLancamento }; 
    };
    
    var getTipo = function(lancamentoConciliado) {
        if(!lancamentoConciliado.tipo) return lancamentoConciliado.tipo;
        return lancamentoConciliado.tipo.id; 
    };
    
    var getDataCompetencia = function(lancamentoConciliado) {
        return lancamentoConciliado.dataCompetencia || moment(); 
    };
    
    var getDataEmissao = function(lancamentoConciliado) {
        return lancamentoConciliado.dataEmissao || moment(); 
    };
    
    var getDataLancamento = function(lancamentoConciliado) {
        return lancamentoConciliado.dataLancamento || moment(); 
    };
    
    var getValor = function(lancamentoConciliado) {
        return lancamentoConciliado.valor || 0;
    };
    
    var getNumeroLote = function(lancamentoConciliado) {
        return lancamentoConciliado.numeroLoteConciliado || null;
    };
    
    var getHistorico = function(lancamentoConciliado) {
        return lancamentoConciliado.historico || null;
    };
    
    var getUsuario = function(lancamentoConciliado) {
        return lancamentoConciliado.usuario || null;
    };

    return {
        handle: handle,
        handleList: handleList
    }

});